server/sonar-server/src/main/java/org/sonar/server/rule/RuleCreator.java

```diff
@@ -166,13 +166,13 @@ private RuleKey createCustomRule(RuleKey ruleKey, NewCustomRule newRule, RuleDto
       .setDescriptionFormat(Format.MARKDOWN)
       .setSeverity(newRule.severity())
       .setStatus(newRule.status())
+      .setType(newRule.type() == null ? templateRuleDto.getType() : newRule.type().getDbConstant())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "type" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)